### DISCUSSION TITLE
aws 3s paginator

### DISCUSSION THREAD

**Discussion by songjiyeong** [grafana/grafana-infinity-datasource#586] [Ideas]
I am using aws 3s to import buckets and objects. Unfortunately this can only fetch up to 1000 objects at a time. I can request again after getting response, but I want to get all objects in one request. Are there any plans to modify s3 to use Paginator-related features? Is the Infinity Plugin completely open source? It took a lot of time to set up the development environment using the Grafana create-plugin tool. Can you give me some tips if I can modify the Infinity Plugin Source to build it?

thank you

Thanks & best regards

**Comment by yesoreyeram** [comment]
It's Apache 2 licensed.
